src/modes/correlation/IndicatorCorrelationChart.svelte

```diff
@@ -0,0 +1,309 @@
+<script>
+  import Vega from '../../components/Vega.svelte';
+  import { BASE_SPEC, guessTopPadding, joinTitle } from '../../specs/commonSpec';
+  import { genCreditsLayer } from '../../specs/lineSpec';
+  import { isMobileDevice } from '../../stores';
+  import Toggle from '../mobile/Toggle.svelte';
+  import CorrelationTooltip from './CorrelationTooltip.svelte';
+  import DownloadMenu from '../mobile/components/DownloadMenu.svelte';
+  import { formatDateISO } from '../../formats';
+  /**
+   * @type {import("../../stores/params").SensorParam}
+   */
+  export let primary;
+  /**
+   * @type {import("../../stores/params").SensorParam}
+   */
+  export let secondary;
+
+  export let lag = 0;
+  /**
+   * @type {Promise<import("../../data/correlation").Lag>}
+   */
+  export let lagData;
+
+  /**
+   * @param {Promise<import("../../data/correlation").Lag>} lag
+   */
+  function prepareData(lag) {
+    return lag.then((lagObj) => {
+      return lagObj.a.map((xi, i) => {
+        const yi = lagObj.b[i];
+        return {
+          x: xi.value,
+          x_date: xi.date_value,
+          x_entry: xi,
+          y: yi.value,
+          y_date: yi.date_value,
+          y_enty: yi,
+        };
+      });
+    });
+  }
+
+  $: data = prepareData(lagData);
+
+  function prepareDownloadRow(row) {
+    const r = {};
+    r.regionId = row.x_entry.propertyId;
+    r.regionLevel = row.x_entry.level;
+    r.regionName = row.x_entry.displayName;
+
+    r.x_indicatorDataSource = primary.value.id;
+    r.x_indicatorId = primary.value.signal;
+    r.x_indicatorName = primary.name;
+    r.x = row.x;
+    r.x_date = formatDateISO(row.x_date);
+
+    r.y_indicatorDataSource = secondary.value.id;
+    r.y_indicatorId = secondary.value.signal;
+    r.y_indicatorName = secondary.name;
+    r.y = row.y;
+    r.y_date = formatDateISO(row.y_date);
+
+    return r;
+  }
+
+  function makeExpression(title, variable, signal) {
+    if (Array.isArray(title)) {
+      return `[${title.map((d) => makeExpression(d, variable, signal)).join(',')}]`;
+    }
+    if (!title.includes(variable)) {
+      return `"${title}"`;
+    }
+    return title
+      .split(variable)
+      .map((d) => `"${d}"`)
+      .join(` + ${signal} + `);
+  }
+
+  function makeIndicatorCompareSpec(primary, secondary, { zero = true, isMobile } = {}) {
+    const title = joinTitle([`${secondary.name} correlated with`, `${primary.name} lagged by $lag days`], isMobile);
```

Comment:
re swapping: however, the axis itself won't swap. I would prefer a version where the title order is the same since the user can easily browse the different lags.

Reply:
Could the text be rewritten to use "correlated with [signal] $lag days earlier" or "$lag days later", depending on the sign of the lag? Saying "days earlier" or "days later" would still be easier to read, since I'm never sure how to interpret the sign of the lag.

Reply:
![image](https://user-images.githubusercontent.com/4129778/114190170-01e5c680-9919-11eb-89cf-7ba007af3dcc.png)

![image](https://user-images.githubusercontent.com/4129778/114190196-0c07c500-9919-11eb-8e03-6ad241aaf384.png)

![image](https://user-images.githubusercontent.com/4129778/114190378-3d809080-9919-11eb-8d86-4e076930b7b9.png)

Reply:
That looks great, thanks.